### ISSUE TITLE
Ollama Embed RAG 500 Bad Request

### ISSUE THREAD

**Issue by YuanfengZhang** [langgenius/dify#8128]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.7.3

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1. Pull [bge-m3](https://ollama.com/library/bge-m3) or [nomic-embed-text](https://ollama.com/library/nomic-embed-text) using Ollama.
2. Register the model above in Dify.
3. Create a new knowledge base, upload a bunch of scientific ariticles. (See attachment)
[Chen et al - 2021 - A multicenter study benchmarking single-cell RNA s.pdf](https://github.com/user-attachments/files/16926290/Chen.et.al.-.2021.-.A.multicenter.study.benchmarking.single-cell.RNA.s.pdf)
[Ahlmann-Eltze and Huber - 2023 - Comparison of transformations for single-cell RNA-seq data.pdf](https://github.com/user-attachments/files/16926292/Ahlmann-Eltze.and.Huber.-.2023.-.Comparison.of.transformations.for.single-cell.RNA-seq.data.pdf)

4. Index them. parameters:
  + Chunk settings: Automatic
  + Index mode: High Quality
  + Embedding model: bge-m3
  + Q&A format: ON
  + Retrieval setting: Hybrid

ENV: Ubuntu 24.04 & NVIDIA A100 * 1

How to solve: The same error has been reported at [Flowise](https://github.com/FlowiseAI/Flowise/issues/1673#issuecomment-2159216044). I tried their solution and it works now.

To sum up, the https://github.com/langgenius/dify/blob/a15791e788ec96ae86ba96a43fee4768b71e9a95/api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py#L87-L93

should be modified as below:
```python
        response = requests.post(
            endpoint_url,
            headers=headers,
            data=json.dumps(payload),
            timeout=(10, 300),
            options={"use_mmap": "true"}
        )
```

### ✔️ Expected Behavior

All files get indexed and embeded.

### ❌ Actual Behavior

I saw:
  + **all of the files failed**
  + **the ollama is running forever, the GPU is at its full speed**
  + Error Message: "api/embed 500 Bad Request"

**Comment by YuanfengZhang** [comment]
Thank u, robot buddy, the face-to-face copying my solution is quite wild.
Anyway, the reason why I open this issue instead of pulling it directly is that I am still not sure it's only my personal problem or it does influence many other people.
Hope you guys have a look at this issue if available. Thx.

**Comment by laipz8200** [comment]
It's looking good, thank you for the report.

**Comment by Firehdx** [comment]
same issue, thx
